js/Extensions/RangeSelector.js

```diff
@@ -1056,16 +1110,29 @@ var RangeSelector = /** @class */ (function () {
             }
             // #10404 - move hide and blur outside focus
             rangeSelector.hideInput(name);
+            rangeSelector.setInputValue(name);
             input.blur(); // #4606
         };
+        var keyDown = false;
         // handle changes in the input boxes
-        input.onchange = updateExtremes;
+        input.onchange = function () {
+            updateExtremes();
+            // Blur input when clicking date input calendar
+            if (!keyDown) {
+                rangeSelector.hideInput(name);
+                input.blur();
+            }
```

Comment:
Unfortunately, this also hides the input when we're arrowing through dates. To reproduce, click the date box so the input appears. Now using the left and right arrows works. Up and down arrows correctly sets the date for the chart, but it hides the input which it shouldn't do. Ideally we should be able to arrow up and down through dates, and the chart should change and input stay visible.

Reply:
Ah yes, need the keydown event then probably.